src/V3Assert.cpp

```diff
@@ -328,20 +330,34 @@ class AssertVisitor final : public VNVisitor {
                     }
                     // Empty case means no property
                     if (!propp) propp = new AstConst{nodep->fileline(), AstConst::BitFalse{}};
-
+                    const string caseTypeStr = nodep->parallelPragma() ? "synthesis parallel_case"
+                                               : nodep->uniquePragma() ? "unique case"
+                                                                       : "unique0 case";
                     const bool allow_none = has_default || nodep->unique0Pragma();
-                    AstNodeExpr* const ohot
-                        = (allow_none ? static_cast<AstNodeExpr*>(
-                               new AstOneHot0{nodep->fileline(), propp})
-                                      : static_cast<AstNodeExpr*>(
-                                          new AstOneHot{nodep->fileline(), propp}));
-                    AstIf* const ifp = new AstIf{
-                        nodep->fileline(), new AstLogNot{nodep->fileline(), ohot},
-                        newFireAssert(nodep,
-                                      "synthesis parallel_case, but multiple matches found")};
-                    ifp->isBoundsCheck(true);  // To avoid LATCH warning
-                    ifp->branchPred(VBranchPred::BP_UNLIKELY);
-                    nodep->addNotParallelp(ifp);
+                    // The following assertion lools as below.
+                    // if (!$onehot(propp)) begin
+                    //     if (propp == '0) begin if (!allow_none) $error("none match"); end
+                    //     else $error("multiple match");
+                    // end
+                    AstNodeExpr* const ohot = new AstOneHot{nodep->fileline(), propp};
+                    AstIf* const ohotIfp
+                        = new AstIf{nodep->fileline(), new AstLogNot{nodep->fileline(), ohot}};
+                    AstIf* const zeroIfp
+                        = new AstIf{nodep->fileline(),
+                                    new AstLogNot{nodep->fileline(), propp->cloneTreePure(false)}};
+                    AstNodeExpr* const exprp = nodep->exprp();
+                    const string valFmt = cvtToStr(exprp->dtypep()->widthMin()) + "'h%X";
+                    if (!allow_none)
+                        zeroIfp->addThensp(
+                            newFireAssert(nodep, caseTypeStr + ", but none matched for " + valFmt,
+                                          exprp->cloneTreePure(false)));
+                    zeroIfp->addElsesp(newFireAssert(
+                        nodep, caseTypeStr + ", but multiple matches found for " + valFmt,
+                        exprp->cloneTreePure(false)));
```

Comment:
In warnings we usually try to single quote text that refers to the specific value/variable.  Also update tests appropriately, thanks.

```suggestion
                            newFireAssert(nodep, caseTypeStr + ", but none matched for '" + valFmt + "'", 
                                          exprp->cloneTreePure(false)));
                    zeroIfp->addElsesp(newFireAssert(
                        nodep, caseTypeStr + ", but multiple matches found for '" + valFmt  + "'",
                        exprp->cloneTreePure(false)));
```